model/service/StructuredVariablesToItemStateService.php

```diff
@@ -0,0 +1,133 @@
+use Psr\Log\LoggerInterface;
+use taoResultServer_models_classes_ResponseVariable as ResponseVariable;
+
+class StructuredVariablesToItemStateService
+{
+    private const RESULT_KEY_URI = 'uri';
+    private const RESULT_KEY_ATTEMPT = 'attempt';
+    private const RESULT_KEY_RESPONSE_VARIABLE = 'taoResultServer_models_classes_ResponseVariable';
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function format(array $testResultVariables): array
+    {
+        $formatted = [];
+        foreach ($testResultVariables as $resultVariables) {
+            $itemResults = $this->formatResultVariables($resultVariables);
+            if ($itemResults) {
+                $itemUri = $resultVariables[self::RESULT_KEY_URI];
+                $formatted[$itemUri][$resultVariables[self::RESULT_KEY_ATTEMPT]] = $itemResults;
+            }
+        }
+
+        return $formatted;
+    }
+
+    private function formatResultVariables(array $resultVariables): ?array
+    {
+        if (false === isset($resultVariables[self::RESULT_KEY_URI], $resultVariables[self::RESULT_KEY_ATTEMPT])) {
+            $this->logNotHaveUriOrAttempt($resultVariables);
+            return null;
+        }
+
+        return $this->formatResponseVariables($resultVariables);
+    }
+
+    private function formatResponseVariables(array $resultVariables): ?array
+    {
+        $responseVariableClasses = $resultVariables[self::RESULT_KEY_RESPONSE_VARIABLE] ?? null;
+        if (false === is_array($responseVariableClasses)) {
+            $this->logNotHaveResponseVariable($resultVariables);
+            return null;
+        }
+
+        $variables = [];
+
+        foreach ($resultVariables[self::RESULT_KEY_RESPONSE_VARIABLE] as $resultVariable) {
+            $responseVariableClass = is_array($resultVariable) ? $resultVariable : [];
+            $responseVariable = $responseVariableClass['var'] ?? null;
+            if ($responseVariable instanceof ResponseVariable) {
+                $variables[$responseVariable->getIdentifier()] = $this->formatResponseVariable(
+                    $responseVariable,
+                    $responseVariableClass
+                );
+            }
+        }
+
+        return $variables;
+    }
+
+    private function formatResponseVariable(ResponseVariable $responseVariable, array $responseVariableClass): array
+    {
+        if ($responseVariable->getBaseType() === 'file') {
+            return $this->getFormatResponse([
+                'base' => [
+                    'file' => [
+                        'uri' => $responseVariableClass['uri'] ?? null
+                    ]
+                ]
+            ]);
+        }
+
+        return $this->getFormatResponse(ResponseVariableFormatter::formatVariableToPci($responseVariable));
+    }
+
+    private function getFormatResponse(array $response): array
```

Comment:
Why have you extracted this method to separate method? You do not perform any business logic. It just format php array.

Reply:
If you notice the same format is repeated twice, but if you think it is useless, I remove the method.